### PR TITLE
Initial splitting message bus into separate assembly

### PR DESCRIPTION
**Pull request by davidfowl**
- Added Microsoft.AspNet.SignalR.Messaging.Tests assembly
- Added shared code to Microsoft.AspNet.SignalR.Common

### CODE REVIEW

File: src/Microsoft.AspNet.SignalR.Messaging/TopicLookup.cs [hunk 1]
```diff
@@ -6,24 +6,19 @@
...
 using System.Collections.Generic;
-using System.Linq;
 using Microsoft.AspNet.SignalR.Infrastructure;
 
 namespace Microsoft.AspNet.SignalR.Messaging
 {
     public sealed class TopicLookup : IEnumerable<KeyValuePair<string, Topic>>
     {
-        // General topics
-        private readonly ConcurrentDictionary<string, Topic> _topics = new ConcurrentDictionary<string, Topic>();
```
Comment (davidfowl) [inline src/Microsoft.AspNet.SignalR.Messaging/TopicLookup.cs]: I can't remember why we split these, /cc @GrabYourPitchforks @DamianEdwards

File: test/Microsoft.AspNet.SignalR.Messaging.Tests/MessageBusFacts.cs [hunk 2]
```diff
@@ -180,6 +168,13 @@ public void SubscribingTopicAfterNoSubscriptionsStateSetsStateToHasSubscription(
             }
         }
 
+        private static TimeSpan TopicTtl(TimeSpan disconnectTimeout)
```
Comment (BrennanConroy) [inline test/Microsoft.AspNet.SignalR.Messaging.Tests/MessageBusFacts.cs]: Change this to caps too? TopicTTL

File: src/Microsoft.AspNet.SignalR.Messaging/Resources.resx [hunk 3]
```diff
@@ -0,0 +1,273 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (halter73) [inline src/Microsoft.AspNet.SignalR.Messaging/Resources.resx]: Are we going to eventually split this up?

File: src/Microsoft.AspNet.SignalR.Messaging/MessageBusOptions.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace Microsoft.AspNet.SignalR.Messaging
+{
+    public class MessageBusOptions
+    {
+        public int MaxTopicsWithNoSubscriptions { get; set; }
+        public int MessageBufferSize { get; set; }
+        public TimeSpan TopicTTL { get; set; }
```
Comment (halter73) [inline src/Microsoft.AspNet.SignalR.Messaging/MessageBusOptions.cs]: So, by default, we destroy topics as soon as their subscription count goes to zero?

File: src/Microsoft.AspNet.SignalR.Messaging/StringMinifier.cs [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 namespace Microsoft.AspNet.SignalR.Infrastructure
 {
-    internal class StringMinifier : IStringMinifier
+    public class StringMinifier : IStringMinifier
```
Comment (halter73) [inline src/Microsoft.AspNet.SignalR.Messaging/StringMinifier.cs]: You're right. This class definitely needs to be public. So beautiful :smile:

File: test/Microsoft.AspNet.SignalR.Messaging.Tests/MessageBusFacts.cs (hunk 2 above)
Comment (halter73) [inline test/Microsoft.AspNet.SignalR.Messaging.Tests/MessageBusFacts.cs]: You might even call it something like `TopicTTLFromDisconnectTimeout`. As it stands, it looks like you are setting the `TopicTTL` to 6 seconds in the test code.

File: src/Microsoft.AspNet.SignalR.Messaging/IStringMinifier.cs [hunk 6]
```diff
@@ -1,4 +1,3 @@
-// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
```
Comment (BrennanConroy) [inline src/Microsoft.AspNet.SignalR.Messaging/IStringMinifier.cs]: Why?

**Pull request merged**